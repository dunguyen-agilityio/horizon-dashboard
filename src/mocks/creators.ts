export const MOCK_CREATORS = [
  {
    id: '1',
    avatar: '/default-avatar.webp',
    userName: 'maddison_c21',
    artwork: 9821,
    rating: 90,
  },
  {
    id: '2',
    avatar: '/default-avatar.webp',
    userName: 'karl.willO2',
    artwork: 7032,
    rating: 80,
  },
  {
    id: '3',
    avatar: '/default-avatar.webp',
    userName: 'andreea.1z',
    artwork: 5204,
    rating: 60,
  },
];
