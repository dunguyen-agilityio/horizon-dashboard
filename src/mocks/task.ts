// Types
import { LABEL, STATUS, Task } from '@/models/Task';

// Mocks
import { MOCK_USERS } from './user';

export const MOCK_TASKS: Task[] = [
  {
    id: '1',
    createdAt: 'Fri, 09 Aug 2024 03:25:32 GMT',
    description:
      "It usually displays this message when you close an unsaved page when you do it on purpose, and it's getting frustrated to see this every time.",
    labels: [LABEL.DONE],
    status: STATUS.IN_PROGRESS,
    title: 'Option to use local/server version feature',
    updatedAt: 'Fri, 09 Aug 2024 03:25:32 GMT',
    cover: '/default-task-cover.png',
    assignees: MOCK_USERS,
  },
  {
    id: '2',
    createdAt: 'Fri, 09 Aug 2024 03:25:32 GMT',
    description:
      "Website Design: The ability to add/modify your own CSS-Selectors like it's done in Venus.",
    labels: [LABEL.DONE],
    status: STATUS.DONE,
    title: 'Add/modify your own CSS-Selectors',
    updatedAt: 'Fri, 09 Aug 2024 03:25:32 GMT',
    assignees: MOCK_USERS,
  },
  {
    id: '3',
    createdAt: 'Fri, 09 Aug 2024 03:25:32 GMT',
    description:
      'When you save some sections as a template and then paste a shortcode to a new page, the layout is broken, some styles are missing - in the editor.',
    labels: [LABEL.ERRORS, LABEL.PENDING],
    status: STATUS.BACKLOG,
    title: 'Shortcode for templates to display correctly',
    updatedAt: 'Fri, 09 Aug 2024 03:25:32 GMT',
    cover: '/default-task-cover.png',
    assignees: MOCK_USERS,
  },
  {
    id: '4',
    createdAt: 'Fri, 09 Aug 2024 03:25:32 GMT',
    description: 'Create your Dashboard with Drag & Drop technique.',
    labels: [LABEL.UPDATES],
    status: STATUS.IN_PROGRESS,
    title: '[UX Design] - Set the default Library tab',
    updatedAt: 'Fri, 09 Aug 2024 03:25:32 GMT',
    assignees: MOCK_USERS,
    cover: '/default-task-cover.png',
  },
  {
    id: '5',
    createdAt: 'Fri, 09 Aug 2024 03:25:32 GMT',
    description: 'Create your Dashboard with Drag & Drop technique.',
    labels: [LABEL.UPDATES],
    status: STATUS.DONE,
    title: '[UX Design] - Set the default Library tab',
    updatedAt: 'Fri, 09 Aug 2024 03:25:32 GMT',
    assignees: MOCK_USERS,
  },
  {
    id: '6',
    createdAt: 'Fri, 09 Aug 2024 03:25:32 GMT',
    description: 'Create your Dashboard with Drag & Drop technique.',
    labels: [LABEL.UPDATES],
    status: STATUS.BACKLOG,
    title: '[UX Design] - Set the default Library tab',
    updatedAt: 'Fri, 09 Aug 2024 03:25:32 GMT',
    assignees: MOCK_USERS,
    cover: '/default-task-cover.png',
  },
];
