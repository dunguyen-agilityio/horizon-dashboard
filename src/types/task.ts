import { Task } from '@/models/Task';

export type TaskFormData = Pick<Task, 'title' | 'status'>;
