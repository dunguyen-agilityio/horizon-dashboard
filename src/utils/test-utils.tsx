import Providers from '@/app/providers';
import { render, RenderOptions } from '@testing-library/react';

const customRender = (ui: React.ReactNode, options?: RenderOptions) =>
  render(ui, { wrapper: Providers, ...options });

export * from '@testing-library/react';

export { customRender as render };
