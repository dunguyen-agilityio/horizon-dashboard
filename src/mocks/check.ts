import { TCheck } from '@/types/check';

export const MOCK_CHECKS: TCheck[] = [
  {
    id: '1',
    name: 'Tony Reichert',
    progress: 45,
    quantity: 20,
    createdAt: new Date('2024-08-14T08:25:04.188Z'),
  },
  {
    id: '2',
    name: 'Zoey Lang',
    progress: 15,
    quantity: 20,
    createdAt: new Date('2024-08-14T08:25:04.188Z'),
  },
  {
    id: '3',
    name: 'Jane Fisher',
    progress: 55,
    quantity: 28,
    createdAt: new Date('2024-08-14T08:25:04.188Z'),
  },
  {
    id: '4',
    name: 'William Howard',
    progress: 65,
    quantity: 30,
    createdAt: new Date('2024-08-14T08:25:04.188Z'),
  },
];
