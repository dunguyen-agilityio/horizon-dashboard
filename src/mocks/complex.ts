import { Complex, ComplexStatus } from '@/types/complex';

export const MOCK_COMPLEXES: Complex[] = [
  {
    id: '1',
    name: 'Tony Reichert',
    progress: 45,
    status: ComplexStatus.APPROVED,
    createdAt: new Date('2024-08-14T08:25:04.188Z'),
  },
  {
    id: '2',
    name: 'Zoey Lang',
    progress: 15,
    status: ComplexStatus.DISABLE,
    createdAt: new Date('2024-08-14T08:25:04.188Z'),
  },
  {
    id: '3',
    name: 'Jane Fisher',
    progress: 55,
    status: ComplexStatus.ERROR,
    createdAt: new Date('2024-08-14T08:25:04.188Z'),
  },
  {
    id: '4',
    name: 'William Howard',
    progress: 65,
    status: ComplexStatus.APPROVED,
    createdAt: new Date('2024-08-14T08:25:04.188Z'),
  },
];
