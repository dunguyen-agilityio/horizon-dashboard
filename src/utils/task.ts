import { MOCK_USERS } from '@/mocks/user';
import { LABEL } from '@/models/Task';
import { CHIP_COLOR } from '@/types/chip';

export const formatStatus = (status: string) =>
  status
    .split('-')
    .map((item) => `${item[0].toUpperCase()}${item.slice(1)}`)
    .join(' ');

export const getColorByLabel = (label: LABEL): CHIP_COLOR => {
  switch (label) {
    case LABEL.ERRORS:
      return 'danger';
    case LABEL.UPDATES:
      return 'primary';
    case LABEL.DONE:
      return 'success';
    case LABEL.PENDING:
      return 'warning';

    default:
      return 'default';
  }
};

export const getUpdatedAssignMembers = (selected: string[]) => {
  return selected
    .map((username) => MOCK_USERS.find((user) => user.username === username))
    .filter((member) => member !== undefined);
};
