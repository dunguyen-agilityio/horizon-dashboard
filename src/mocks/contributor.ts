import { MOCK_IMAGE_RESPONSE, MOCK_ROLE_RESPONSE, MOCK_USERS } from './user';
import { MOCK_NFTS_RESPONSE } from './nft';
import { ContributorResponse } from '@/types/contributor';
import { formatUser } from '@/utils/user';
import { StrapiResponse } from '@/types/strapi';

export const MOCK_CONTRIBUTORS: StrapiResponse<ContributorResponse>[] =
  MOCK_USERS.map((user, index) => ({
    id: String(index),
    attributes: {
      avatar: { data: MOCK_IMAGE_RESPONSE },
      join_nfts: { data: MOCK_NFTS_RESPONSE },
      users_permissions_user: {
        data: {
          id: user.id,
          attributes: {
            ...formatUser(user),
            avatar: { data: MOCK_IMAGE_RESPONSE },
            join_nfts: { data: MOCK_NFTS_RESPONSE },
            role: MOCK_ROLE_RESPONSE,
          },
        },
      },
    },
  }));
