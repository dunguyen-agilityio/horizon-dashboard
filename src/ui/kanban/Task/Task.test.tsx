import { render } from '@/utils/test-utils';

import { MOCK_TASKS } from '@/mocks/task';

import { Task } from '.';

const mockTask = MOCK_TASKS[0];

describe('Task tests', () => {
  it('Should match snapshot', () => {
    const { container } = render(<Task {...mockTask} />);
    expect(container).toMatchSnapshot();
  });
});
