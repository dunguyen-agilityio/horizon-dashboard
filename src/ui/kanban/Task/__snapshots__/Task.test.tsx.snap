// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Task tests Should match snapshot 1`] = `
<div>
  <div
    data-overlay-container="true"
  >
    <script>
      !function(){try{var d=document.documentElement,c=d.classList;c.remove('light','dark');var e=localStorage.getItem('theme');if('system'===e||(!e&&false)){var t='(prefers-color-scheme: dark)',m=window.matchMedia(t);if(m.media!==t||m.matches){d.style.colorScheme = 'dark';c.add('dark')}else{d.style.colorScheme = 'light';c.add('light')}}else if(e){c.add(e|| '')}else{c.add('light')}if(e==='light'||e==='dark'||!e)d.style.colorScheme=e||'light'}catch(e){}}()
    </script>
    <div
      class="flex flex-col gap-[10px] pt-[18px] pl-[23px] pb-[23px] pr-[21px] rounded-[15px] bg-white shadow-light-card dark:shadow-none dark:bg-indigo-light max-w-[466px]"
    >
      <div
        class="flex justify-between"
      >
        <h3
          class="text-primary dark:text-white text-extra font-bold leading-large max-w-[90%]"
        >
          Option to use local/server version feature
        </h3>
        <button
          aria-label="edit-button"
          class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent data-[pressed=true]:scale-[0.97] outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 text-small gap-2 rounded-medium [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none text-default-foreground data-[hover=true]:opacity-hover p-0 min-w-fit h-fit bg-transparent"
          type="button"
        >
          <svg
            data-testid="pencil-icon"
            fill="none"
            height="17"
            viewBox="0 0 17 17"
            width="17"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              clip-path="url(#clip0_101_11790)"
            >
              <path
                d="M2.125 12.3675V14.5208C2.125 14.7192 2.28083 14.875 2.47917 14.875H4.6325C4.72458 14.875 4.81667 14.8396 4.88042 14.7688L12.6154 7.04083L9.95917 4.38458L2.23125 12.1125C2.16042 12.1833 2.125 12.2683 2.125 12.3675ZM14.6696 4.98667C14.9458 4.71042 14.9458 4.26417 14.6696 3.98792L13.0121 2.33042C12.7358 2.05417 12.2896 2.05417 12.0133 2.33042L10.7171 3.62667L13.3733 6.28292L14.6696 4.98667Z"
                fill="#A3AED0"
              />
            </g>
            <defs>
              <clippath
                id="clip0_101_11790"
              >
                <rect
                  fill="white"
                  height="17"
                  width="17"
                />
              </clippath>
            </defs>
          </svg>
        </button>
      </div>
      <div
        class="relative min-h-[285px] min-w-[308px]"
      >
        <img
          alt="cover"
          data-nimg="fill"
          decoding="async"
          loading="lazy"
          sizes="100vw"
          src="/_next/image?url=%2Fdefault-task-cover.webp&w=3840&q=75"
          srcset="/_next/image?url=%2Fdefault-task-cover.webp&w=640&q=75 640w, /_next/image?url=%2Fdefault-task-cover.webp&w=750&q=75 750w, /_next/image?url=%2Fdefault-task-cover.webp&w=828&q=75 828w, /_next/image?url=%2Fdefault-task-cover.webp&w=1080&q=75 1080w, /_next/image?url=%2Fdefault-task-cover.webp&w=1200&q=75 1200w, /_next/image?url=%2Fdefault-task-cover.webp&w=1920&q=75 1920w, /_next/image?url=%2Fdefault-task-cover.webp&w=2048&q=75 2048w, /_next/image?url=%2Fdefault-task-cover.webp&w=3840&q=75 3840w"
          style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
        />
      </div>
      <p
        class="text-secondary dark:text-secondary text-medium"
      >
        It usually displays this message when you close an unsaved page when you do it on purpose, and it's getting frustrated to see this every time.
      </p>
      <div
        class="flex justify-between mt-5 items-center"
      >
        <div
          class="flex items-center justify-center h-auto w-max"
          role="group"
        >
          <span
            class="flex relative justify-center items-center box-border overflow-hidden align-middle z-0 outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 w-10 h-10 text-tiny bg-default text-default-foreground rounded-full ring-2 ring-offset-2 ring-offset-background dark:ring-offset-background-dark data-[hover=true]:-translate-x-3 rtl:data-[hover=true]:translate-x-3 transition-transform data-[focus-visible=true]:-translate-x-3 rtl:data-[focus-visible=true]:translate-x-3 ring-default ms-0"
            tabindex="-1"
          >
            <img
              alt="User 1"
              class="flex object-cover w-full h-full transition-opacity !duration-500 opacity-0 data-[loaded=true]:opacity-100"
              data-nimg="1"
              decoding="async"
              height="20"
              loading="lazy"
              src="/_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdn5vw6fjp%2Fimage%2Fupload%2Fv1726630608%2Fimage_de1e29b0b7.png&w=48&q=75"
              srcset="/_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdn5vw6fjp%2Fimage%2Fupload%2Fv1726630608%2Fimage_de1e29b0b7.png&w=32&q=75 1x, /_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdn5vw6fjp%2Fimage%2Fupload%2Fv1726630608%2Fimage_de1e29b0b7.png&w=48&q=75 2x"
              style="color: transparent;"
              width="20"
            />
          </span>
          <span
            class="flex relative justify-center items-center box-border overflow-hidden align-middle z-0 outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 w-10 h-10 text-tiny bg-default text-default-foreground rounded-full ring-2 ring-offset-2 ring-offset-background dark:ring-offset-background-dark data-[hover=true]:-translate-x-3 rtl:data-[hover=true]:translate-x-3 transition-transform data-[focus-visible=true]:-translate-x-3 rtl:data-[focus-visible=true]:translate-x-3 ring-default -ms-2"
            tabindex="-1"
          >
            <img
              alt="User 2"
              class="flex object-cover w-full h-full transition-opacity !duration-500 opacity-0 data-[loaded=true]:opacity-100"
              data-nimg="1"
              decoding="async"
              height="20"
              loading="lazy"
              src="/_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdn5vw6fjp%2Fimage%2Fupload%2Fv1726630608%2Fimage_de1e29b0b7.png&w=48&q=75"
              srcset="/_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdn5vw6fjp%2Fimage%2Fupload%2Fv1726630608%2Fimage_de1e29b0b7.png&w=32&q=75 1x, /_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdn5vw6fjp%2Fimage%2Fupload%2Fv1726630608%2Fimage_de1e29b0b7.png&w=48&q=75 2x"
              style="color: transparent;"
              width="20"
            />
          </span>
          <span
            class="flex relative justify-center items-center box-border overflow-hidden align-middle z-0 outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 w-10 h-10 text-tiny bg-default text-default-foreground rounded-full ring-2 ring-offset-2 ring-offset-background dark:ring-offset-background-dark data-[hover=true]:-translate-x-3 rtl:data-[hover=true]:translate-x-3 transition-transform data-[focus-visible=true]:-translate-x-3 rtl:data-[focus-visible=true]:translate-x-3 ring-default -ms-2"
            tabindex="-1"
          >
            <img
              alt="User 3"
              class="flex object-cover w-full h-full transition-opacity !duration-500 opacity-0 data-[loaded=true]:opacity-100"
              data-nimg="1"
              decoding="async"
              height="20"
              loading="lazy"
              src="/_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdn5vw6fjp%2Fimage%2Fupload%2Fv1726630608%2Fimage_de1e29b0b7.png&w=48&q=75"
              srcset="/_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdn5vw6fjp%2Fimage%2Fupload%2Fv1726630608%2Fimage_de1e29b0b7.png&w=32&q=75 1x, /_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdn5vw6fjp%2Fimage%2Fupload%2Fv1726630608%2Fimage_de1e29b0b7.png&w=48&q=75 2x"
              style="color: transparent;"
              width="20"
            />
          </span>
          <span
            class="flex relative justify-center items-center box-border overflow-hidden align-middle z-0 outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 w-10 h-10 text-tiny bg-default text-default-foreground rounded-full ring-2 ring-offset-2 ring-offset-background dark:ring-offset-background-dark -ms-2 data-[hover=true]:-translate-x-3 rtl:data-[hover=true]:translate-x-3 transition-transform data-[focus-visible=true]:-translate-x-3 rtl:data-[focus-visible=true]:translate-x-3 ring-default hover:-translate-x-0"
            tabindex="-1"
          >
            <span
              aria-label="+1"
              class="absolute top-1/2 left-1/2 -translate-x-1/2 -translate-y-1/2 font-normal text-center text-inherit"
              role="img"
            >
              +1
            </span>
          </span>
        </div>
        <div
          class="relative max-w-fit min-w-min inline-flex items-center justify-between box-border whitespace-nowrap h-7 bg-success text-white uppercase text-xs leading-[15px] rounded-[10px] py-[6.5px] px-[25px]"
        >
          <span
            class="flex-1 text-inherit font-normal px-2"
          >
            Done
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
